torch/profiler/_pattern_matcher.py

```diff
@@ -603,8 +631,31 @@ def report_all_anti_patterns(prof, should_benchmark: bool = False, json_report_d
             if report_msg not in reported:
                 message_list.append(report_msg)
                 reported.add(report_msg)
+                src_location, line_no = source_code_location(event).split(":")
+                report_dict.setdefault(src_location, []).append({
```

Comment:
TIL you can use the result of `setdefault`. That will definitely simplify my code.

Reply:
💯